docs/en/integrations/data-ingestion/apache-spark/index.md

```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
 In this section, we will provide details on how to
 use the [ClickHouse official JDBC connector](https://github.com/ClickHouse/clickhouse-java) with Spark.
 
+
 ### Read data
 
+<Tabs groupId="spark_apis">
+<TabItem value="Java" label="Java" default>
+
 ```java
 public static void main(String[] args) {
         // Initialize Spark session
         SparkSession spark = SparkSession.builder().appName("example").master("local").getOrCreate();
 
-        // JDBC connection details
-        String jdbcUrl = "jdbc:ch://localhost:8123/default";
+        String jdbcURL = "jdbc:ch://localhost:8123/default";
+        String query = "select * from example_table where id > 2";
+
+
+        //---------------------------------------------------------------------------------------------------
+        // Load the table from ClickHouse using jdbc method
+        //---------------------------------------------------------------------------------------------------
         Properties jdbcProperties = new Properties();
         jdbcProperties.put("user", "default");
         jdbcProperties.put("password", "123456");
 
-        // Load the table from ClickHouse
-        Dataset<Row> df = spark.read().jdbc(jdbcUrl, "example_table", jdbcProperties);
+        Dataset<Row> df1 = spark.read().jdbc(jdbcURL, String.format("(%s)", query), jdbcProperties);
+
+        df1.show();
+
+        //---------------------------------------------------------------------------------------------------
+        // Load the table from ClickHouse using load method
+        //---------------------------------------------------------------------------------------------------
+        Dataset<Row> df2 = spark.read()
+                .format("jdbc")
+                .option("url", jdbcURL)
+                .option("user", "default")
+                .option("password", "123456")
+                .option("query", query)
+                .load();
+
+
+        df2.show();
 
-        // Show the DataFrame
-        df.show();
 
         // Stop the Spark session
         spark.stop();
     }
 ```
+</TabItem>
+<TabItem value="Scala" label="Scala">
+
+```java
+object ReadData extends App {
+  // Initialize Spark session
+  val spark: SparkSession = SparkSession.builder.appName("example").master("local").getOrCreate
+
+  val jdbcURL = "jdbc:ch://localhost:8123/default"
+  val query: String = "select * from example_table where id > 2"
+
+
+  //---------------------------------------------------------------------------------------------------
+  // Load the table from ClickHouse using jdbc method
+  //---------------------------------------------------------------------------------------------------
+  val connectionProperties = new Properties()
+  connectionProperties.put("user", "default")
+  connectionProperties.put("password", "123456")
+
+  val df1: Dataset[Row] = spark.read.
```

Comment:
I think we should write meaningful variables names

Reply:
Do you have any ideas for meaningful variable names? it is a standard practice to call a DataFrame df